### PR TITLE
refactor(js): convert jest.config.js to jest.config.ts

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

refactor(js): convert jest.config.js to jest.config.ts

---

Reference: #201 [comment](https://github.com/logto-io/js/pull/201#discussion_r813719266)

## Linear Issue Reference

LOG-1628
LOG-1600

## Testing

Pass all existing tests.

### CODE REVIEW

File: packages/js/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,2128 @@
+{
```
Comment (demonzoo) [inline packages/js/package-lock.json:1]: Why do we even have this lock file since we are using pnpm?

**Review by gao-sun — CHANGES_REQUESTED** [review]
why `package-lock.json`..

**Comment by IceHe** [comment]
> why `package-lock.json`..

My bad. Removed.

File: packages/js/package-lock.json (hunk 1 above)
Comment (IceHe) [reply packages/js/package-lock.json]: Removed.

**Review by demonzoo — APPROVED** [review]

File: packages/js/src/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import * as core from '.';
```
Comment (gao-sun) [inline packages/js/src/index.test.ts:1]: what's the point of this file?

Comment (IceHe) [reply packages/js/src/index.test.ts:1]: test exports. not necessary enough.
Removed. (See [discussion](https://silverhand-io.slack.com/archives/C02CMB8JVJB/p1645772028319539))

**Review by gao-sun — APPROVED** [review]

**Pull request merged**